server/src/main/java/org/elasticsearch/action/search/SearchScrollAsyncAction.java

```diff
@@ -240,12 +240,7 @@ protected final void sendResponse(
         final AtomicArray<? extends SearchPhaseResult> fetchResults
     ) {
         try {
-            final InternalSearchResponse internalResponse = SearchPhaseController.merge(
```

Comment:
These (couple more spots like this one where we hat the array split across two params) are the most important ones, getting the life-cycle of the fetch results right was one of the trickiest aspects of this.